File: client/src/main/java/de/qabel/desktop/ClientPlugin.java
package de.qabel.desktop;

import de.qabel.core.event.EventSource;
import de.qabel.desktop.inject.CompositeServiceFactory;

public interface ClientPlugin {
    void initialize(CompositeServiceFactory serviceFactory, EventSource events);
    default String getName() {
        return getClass().getName();
    }
}

File: client/src/main/java/de/qabel/desktop/plugin/ClientPlugin.kt
package de.qabel.desktop.plugin

import de.qabel.core.event.EventSource
import de.qabel.desktop.ServiceFactory
import de.qabel.desktop.inject.CompositeServiceFactory

interface ClientPlugin {
    fun initialize(serviceFactory: CompositeServiceFactory, events: EventSource)
    val name: String
        @JvmSynthetic get() = javaClass.name
}
interface ServiceFactoryProvider {
    fun getServiceFactory() : ServiceFactory
}
